internal/app/readycheck.go

```diff
@@ -146,7 +146,8 @@ func (a *Application) RegisterHandler() {
 	mux.HandleFunc("/stop", a.stopEtcdHandler)
 
 	a.server = &http.Server{
-		Addr:    fmt.Sprintf(":%d", ServerPort),
-		Handler: mux,
+		Addr:              fmt.Sprintf(":%d", ServerPort),
+		Handler:           mux,
+		ReadHeaderTimeout: etcdConnectionTimeout,
```

Comment:
Is it necessary to define `ReadHeaderTimeout` ?

Reply:
gosec threw an error here that `ReadHeaderTimeout` is not defined, which can cause malicious requests to block the process with indefinitely large request headers. That's the reason I defined it, set to the standard `etcdConnectionTimeout` of `5s`. I think this is sufficient, because `1s` may be too less, in case of actual read latency for the request. WDYT?

Reply:
> gosec threw an error here that ReadHeaderTimeout is not defined, which can cause malicious requests to block the process with indefinitely large request headers

ok, got it. thanks

> That's the reason I defined it, set to the standard etcdConnectionTimeout of 5s. I think this is sufficient, because 1s may be too less, in case of actual read latency for the request

what about `2sec` ? and new const for it as well since `etcdConnectionTimeout` is meant for etcd connection, seems little misleading as well. wdyt ?

Reply:
Sure, will create a new constant. But I would prefer to keep it at `5s` for now, to avoid unnecessary request timeouts.